components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -71,14 +75,18 @@ class GeckoEngineSession(
         ) { }
 
         override fun onPageStart(session: GeckoSession?, url: String?) {
-            notifyObservers { onProgress(PROGRESS_START) }
-            notifyObservers { onLoadingStateChange(true) }
+            notifyObservers(
+                { onProgress(PROGRESS_START) },
+                { onLoadingStateChange(true) }
+            )
```

Comment:
What's your reason for multiple lambdas and not something like:

```Kotlin
notifyObservers {
  onProgress(PROGRESS_START)
  onLoadingStateChange(true)
}
```

Do you want to propagate the first event to all observers first?

Reply:
Yes, that's what I thought I wanted, but I just realized I didn't implement this correctly. Now, even better, I don't think we need this at all :)

If we want to propagate an event to all observers before the next event is sent we can do it separately (basically how it was):
```
notifyObservers( { onEventA() } )
notifyObservers( { onEventB() } )
```

If we want to notify observers of multiple events in one go:

```
notifyObservers {
  onEventA()
  onEventB()
}
```

So, nothing to see here ;)

Reply:
👍